### PR TITLE
Customer features

### PR DESCRIPTION
**Pull request by zxgoh**
Commented out some test cases since not using for now
App is functional for demo
Refactored the `parseName()` method for dish to `parseDishName()` to avoid overlap with `parseName()` for customer

### CODE REVIEW

**Review by jaegarpoon — APPROVED** [review]
Looks good to me! Will probably need to clean up some of the remaining persons stuff left, probably in the next iteration

File: src/main/java/seedu/address/MainApp.java [hunk 1]
```diff
@@ -173,7 +173,7 @@ public void start(Stage primaryStage) {
 
     @Override
     public void stop() {
-        logger.info("============================ [ Stopping Address Book ] =============================");
+        logger.info("============================ [ Stopping AddressCustomer Book ] =============================");
```
Comment (hieunm1821) [inline src/main/java/seedu/address/MainApp.java:176]: Can I ask what does AddressCustomer mean? I think this edit due to the auto replacing. This issue happens somewhere else.

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 2]
```diff
@@ -38,10 +38,10 @@ public AddCommand parse(String args) throws ParseException {
...
         }
 
-        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
-        Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
-        Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
-        Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
+        Name name = null;
+        Phone phone = null;
+        Email email = null;
+        Address address = null;
```
Comment (hieunm1821) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:44]: I think we will remove these things belong to Person class and the base app eventually. Do these edits affect the app or the test?

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -37,21 +38,37 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
...
      *
      * @throws ParseException if the given {@code name} is invalid.
      */
-    public static Name parseName(String name) throws ParseException {
+    public static Name parseDishName(String name) throws ParseException {
         requireNonNull(name);
         String trimmedName = name.trim();
-        if (!Name.isValidName(trimmedName)) {
-            throw new ParseException(Name.MESSAGE_CONSTRAINTS);
+        if (!NameCustomer.isValidName(trimmedName)) {
```
Comment (hieunm1821) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:65]: Should it be new class named NameDish?

**Review by hieunm1821 — APPROVED** [review]

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java (hunk 2 above)
Comment (zxgoh) [reply src/main/java/seedu/address/logic/parser/AddCommandParser.java:44]: I think we wont be running an `AddCommandParserTest` so maybe we can remove this

File: src/main/java/seedu/address/MainApp.java (hunk 1 above)
Comment (zxgoh) [reply src/main/java/seedu/address/MainApp.java:176]: Oh ya, i accidentally included this in a refactoring, will change it now

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 3 above)
Comment (zxgoh) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: Ok, fixed this error!

**Comment by zxgoh** [comment]
fixes #29 #30

**Pull request merged**